docs/sop/PRODUCTION_DEPLOY.md

```diff
@@ -0,0 +1,344 @@
+expand the Protected branches section for the gigadb-website project.
+
+### Procedure
+
+1. Go to [Gitlab Upstream pipeline page](https://gitlab.com/gigascience/upstream/gigadb-website/-/pipelines)
+and run all the jobs in the staging build stage in your pipeline, including but are not limited to `build_staging`, `TidewaysBuildStaging`.
+
+2. Next, run all the jobs in the staging deploy stage in your pipeline, including but are not limited to `sd_gigadb`, `TidewaysDeployStaging`.
+
+## Provision AWS infrastructure for `live` environment using Terraform and Ansible
+
+You have to make sure there is no error when deploying `staging` environment.
+Then, you need to follow the exact steps in the staging provision section and look into the details of
+`terrafor show` and `terraform plan` commands.
+
+And also, you can click to download the [shared Terraform state for Upstream's live environment](https://gitlab.com/api/v4/projects/11385199/terraform/state/live_infra/versions/76)
+for checking in the latter live deployment steps.
+
+Change directory to the `envs` folder:
+```
+$ cd ops/infrastructure/envs
+```
+
+Change directory to the `live` directory:
+```
+$ cd live
+```
+
+Copy terraform files to `live` environment:
+```
+$ ../../../scripts/tf_init.sh --project gigascience/upstream/gigadb-website --env live
+
+You need to specify an AWS region:
+ap-east-1
+
+You need to specify the path to the ssh private key to use to connect to the EC2 instance: 
+~/.ssh/id-rsa-aws-hk-gigadb.pem
+
+You need to specify your GitLab username:
+pli888 | rijam | kencho51
+
+You need to specify a backup file created by the files-url-updater tool:
+../../../../gigadb/app/tools/files-url-updater/sql/gigadbv3_20210929_v9.3.25.backup
+```
+
+Use Gigadb AWS IAM user account to provision production staging server:
+```
+$ terraform show # will show all the existing resources, which should be the same as the terraform state file `live_infra` from gitlab.
+$ terraform plan # terraform is idempotent and should not try to create new instances for already existing upstream staging, unless the new instance is expected to create.
+$ terraform apply # will make changes to the existing infrastructure and update the terraform state file, input `yest` if the changes are expected to make.
+$ terraform refresh
+```
+
+Copy ansible files into `live` environment:
+```
+$ ../../../scripts/ansible_init.sh --env live
+```
+
+Install third party Ansible roles:
+```
+$ ansible-galaxy install -r ../../../infrastructure/requirements.yml
+```
+
+Provision RDS via bastion server:
+```
+$ env OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES ansible-playbook -i ../../inventories bastion_playbook.yml --extra-vars="gigadb_env=staging"
+```
+
+Enable cronjob for periodically resetting the database:
+```
+$ env OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES ansible-playbook -i ../../inventories bastion_playbook.yml -e "reset_database_cronjob_state=present" --extra-vars="gigadb_env=staging"
```

Comment:
**typo**: we are in the section for provisioning live, but  I see `--extra-vars="gigadb_env=staging"` here
**suggestion**: replace with `--extra-vars="gigadb_env=live"`